### PR TITLE
community[patch]: PGVectorStore: Add dimensions parameter to initialize function

### PR DESCRIPTION
**Pull request by clemenspeters**
When using `createHnswIndex` the index is created in the database, but if the vector column doesn't have a fixed dimension, the query planner doesn't make use of the index and the query is still slow.

Example usage:
```
  const dimensions = 1536;
  const pgVectorStore = await PGVectorStore.initialize(embeddings, pgConfig, dimensions)

  await pgVectorStore.createHnswIndex({
    dimensions,
    efConstruction: 64,
    m: 16,
  });
  ```
  
  @PetersClemens

### CODE REVIEW

File: libs/langchain-community/src/vectorstores/pgvector.ts [hunk 1]
```diff
@@ -164,16 +164,18 @@ export class PGVectorStore extends VectorStore {
...
    * @param embeddings - Embeddings instance.
    * @param fields - `PGVectorStoreArgs` instance.
+   * @param dimensions Number of dimensions in your vector data type. For example, use 1536 for OpenAI's `text-embedding-3-small`. If not set, indexes like HNSW might not be used during query time.
    * @returns A new instance of `PGVectorStore`.
    */
   static async initialize(
     embeddings: EmbeddingsInterface,
-    config: PGVectorStoreArgs
+    config: PGVectorStoreArgs,
+    dimensions?: number
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/pgvector.ts:173]: I'll move this into `PGVectorStoreArgs` if that's ok

**Comment by jacoblee93** [comment]
Hey @clemenspeters, are you ok with this instead?

https://github.com/langchain-ai/langchainjs/pull/6413

If so feel free to patch onto your PR

**Comment by jacoblee93** [comment]
Actually since cutting a community release now - just merged https://github.com/langchain-ai/langchainjs/pull/6413

Thank you for this!

**Comment by clemenspeters** [comment]
I guess the links were meant to https://github.com/langchain-ai/langchainjs/pull/6452
Thanks for integrating / improving and merging 👍

**Pull request closed without merging**